android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java

```diff
@@ -53,7 +53,7 @@ public boolean touchDown (int screenX, int screenY, int pointer, int button) {
     public boolean touchUp (int screenX, int screenY, int pointer, int button) {
         if (screenTouchedMarker.epsilonEquals(screenX, screenY, SENSITIVITY)) {
             ModelInstance selectedPart = getSelectedPartId(screenX, screenY);
-            if (selectedPart != null) {
+            if (selectedPart != null && !selectedPart.materials.get(0).id.equals("landing-gear")) {
```

Comment:
landing-gear? In a car?! This is also very model-specific, should be extracted, removed, or at least commented what's it for.

Reply:
Hardcoding a single invalid id which happened to be present in the current model is not a good idea.  A better check would be to verify if the id is on the list of valid ids listed in JSON schema.

Reply:
changed